components/nimbus/src/stateless/cirrus_client.rs

```diff
@@ -86,35 +97,45 @@ impl CirrusClient {
     /// method. The information returned from this method can be used to merge the default feature
     /// values with the values applied by the enrolled experiments and to send enrollment-
     /// related Glean events.
-    pub fn handle_enrollment(&self, request: EnrollmentRequest) -> Result<EnrollmentResponse> {
+    pub fn handle_enrollment(&self, request: JsonObject) -> Result<JsonObject> {
         let EnrollmentRequest {
             client_id,
-            context,
+            app_context,
+            request_context,
             is_user_participating,
             next_experiments,
             prev_enrollments,
-        } = request;
+        } = from_value(Value::Object(request))?;
+        let app_context: AppContext = from_value(Value::Object(app_context)).unwrap();
+        let request_context: RequestContext = from_value(Value::Object(request_context)).unwrap();
         let client_id = if let Some(client_id) = client_id {
             client_id
         } else {
             return Err(NimbusError::CirrusError(
                 CirrusClientError::RequestMissingParameter("client_id".to_string()),
             ));
         };
+        let next_experiments: Vec<Experiment> = from_value(Value::Array(next_experiments)).unwrap();
+        let prev_enrollments: Vec<ExperimentEnrollment> =
+            from_value(Value::Array(prev_enrollments)).unwrap();
```

Comment:
Nit: a comment here re: why we're doing this in Rust, rather than in Uniffi-generated code.